src/status_im2/contexts/activity_center/view.cljs

```diff
@@ -123,47 +124,59 @@
 
-(defn render-notification
-  [{:keys [type] :as notification} index]
-  [rn/view {:style (style/notification-container index)}
-   (cond
-     (= type types/contact-verification)
-     [contact-verification/view notification {}]
+(defn notification-component
+  [_ _ _ _]
+  (let [height               (atom 0)
+        set-swipeable-height #(reset! height (oops/oget % "nativeEvent.layout.height"))]
+    (fn [{:keys [type] :as notification} index _ active-swipeable]
+      (let [swipeable-args {:height           height
+                            :active-swipeable active-swipeable
+                            :notification     notification}]
+        [rn/view {:style (style/notification-container index)}
+         (cond
+           (= type types/contact-verification)
+           [contact-verification/view notification {}]
 
-     (= type types/contact-request)
-     [contact-request/view notification]
+           (= type types/contact-request)
+           [contact-request/swipeable swipeable-args
+            [contact-request/view notification set-swipeable-height]]
 
-     (= type types/mention)
-     [mentions/view notification]
+           (= type types/mention)
+           [mentions/swipeable swipeable-args
+            [mentions/view notification set-swipeable-height]]
 
-     (= type types/reply)
-     [reply/view notification]
+           (= type types/reply)
+           [reply/swipeable swipeable-args
+            [reply/view notification set-swipeable-height]]
 
-     (= type types/admin)
-     [admin/view notification]
+           (= type types/admin)
+           [admin/swipeable swipeable-args
+            [admin/view notification set-swipeable-height]]
 
-     (some types/membership [type])
-     [membership/view notification]
+           (some types/membership [type])
+           [membership/view notification]
 
-     :else
-     nil)])
+           :else
+           nil)]))))
 
 (defn view
   []
-  [:f>
-   (fn []
-     (react/effect! #(rf/dispatch [:activity-center.notifications/fetch-first-page]))
-     [safe-area/consumer
-      (fn [{:keys [top bottom]}]
-        (let [notifications (rf/sub [:activity-center/filtered-notifications])
-              window-width  (rf/sub [:dimensions/window-width])]
-          [rn/view {:style (style/screen-container window-width top bottom)}
-           [header]
-           [rn/flat-list
-            {:data                      notifications
-             :content-container-style   {:flex-grow 1}
-             :empty-component           [empty-tab]
-             :key-fn                    :id
-             :on-scroll-to-index-failed identity
-             :on-end-reached            #(rf/dispatch [:activity-center.notifications/fetch-next-page])
-             :render-fn                 render-notification}]]))])])
+  (let [active-swipeable (atom nil)]
+    [:f>
+     (fn []
+       (react/effect! #(rf/dispatch [:activity-center.notifications/fetch-first-page]))
```

Comment:
I think we should use `rn/use-effect` or `rn/use-effect-once`

Reply:
Thanks, I will double-check.

Reply:
Done, much better now.

Reply:
I believe we currently have a bug in `rn/use-effect-once` where you have to return `js/undefined` explicitly. I see you are not returning `undefined` so maybe double check it is working properly.

Reply:
I tested manually, it's working, but good reminder. I'll add an explicit return for js/undefined.

Reply:
I don't know if somebody fixed the problem recently, but we can see `js/undefined` is explicitly returned. Please, take a look:

```clojure
(defn use-effect
  ([effect-fn]
   (use-effect effect-fn []))
  ([effect-fn deps]
   (react/useEffect
    #(let [ret (effect-fn)]
       (if (fn? ret) ret js/undefined))
    (bean/->js deps))))

(def use-callback react/useCallback)

(defn use-effect-once
  [effect-fn]
  (use-effect effect-fn))
```

Reply:
@briansztamfater fixed this

Reply:
Awesome then!